model/time_series_update.go

```diff
@@ -0,0 +1,236 @@
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTestNameKey): bson.M{"$exists": true},
+			},
+		},
+		{
+			"$match": bson.M{
+				"$expr": bson.M{
+					"$lt": []string{
+						"$" + bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisProcessedAtKey),
+						"$" + bsonutil.GetDottedKeyName(perfRollupsKey, perfRollupsProcessedAtKey),
+					},
+				},
+			},
+		},
+		{
+			"$group": bson.M{
+				"_id": bson.M{
+					"project": "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey),
+					"variant": "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVariantKey),
+					"task":    "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTaskNameKey),
+					"test":    "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTestNameKey),
+				},
+			},
+		},
+		{
+			"$replaceRoot": bson.M{
+				"newRoot": "$_id",
+			},
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get metrics needing change point detection")
+	}
+	defer cur.Close(ctx)
+	var res []PerformanceResultSeriesID
+	err = cur.All(ctx, &res)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to decode metrics needing change results")
+	}
+	return res, nil
+}
+
+func GetPerformanceResultSeriesIDs(ctx context.Context, env cedar.Environment) ([]PerformanceResultSeriesID, error) {
+	cur, err := env.GetDB().Collection(perfResultCollection).Aggregate(ctx, []bson.M{
+		{
+			"$match": bson.M{
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoOrderKey):    bson.M{"$exists": true},
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoMainlineKey): true,
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey):  bson.M{"$exists": true},
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVariantKey):  bson.M{"$exists": true},
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTaskNameKey): bson.M{"$exists": true},
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTestNameKey): bson.M{"$exists": true},
+			},
+		},
+		{
+			"$unwind": "$" + bsonutil.GetDottedKeyName(perfRollupsKey, perfRollupsStatsKey),
+		},
+		{
+			"$group": bson.M{
+				"_id": bson.M{
+					"project": "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey),
+					"variant": "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVariantKey),
+					"task":    "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTaskNameKey),
+					"test":    "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTestNameKey),
+					"args":    "$" + bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoArgumentsKey),
+				},
+			},
+		},
+		{
+			"$replaceRoot": bson.M{
+				"newRoot": "$_id",
+			},
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "Cannot aggregate time series ids")
+	}
+	defer cur.Close(ctx)
+	var res []PerformanceResultSeriesID
+	err = cur.All(ctx, &res)
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not decode time series ids")
+	}
+	return res, nil
+}
```

Comment:
Add space (are you using the go formatter?)

Reply:
Yeah I was using `go fmt`. Weird it wasn't adding a space. Added it now.